### ISSUE TITLE
no-check does not support `import =` namespace alias

### ISSUE THREAD

**Issue by lts20050703** [denoland/deno#12395]
```ts
// deps.ts
export * as mongo from "https://deno.land/x/mongo@v0.27.0/mod.ts"
```
```ts
// mongo.ts
import { mongo } from "./deps.ts"
import MongoClient = mongo.MongoClient
const mongoClient = new MongoClient()
```
`deno run -A --unstable --no-check mod.ts`
```
error: Uncaught ReferenceError: MongoClient is not defined
const mongoClient = new MongoClient()
                    ^
    at file:///home/lts372005/Desktop/discordeno-kingdoms/mongo.ts:3:21
```
# Why I `import Collection = mongo.Collection` and not `const Collection = mongo.Collection` (or in the case of denoland/deno_lint#934, why not `type Collection = mongo.Collection`
Read https://github.com/microsoft/TypeScript/issues/2552#issuecomment-87893040
TL;DR Consider the following
```ts
import { mongo } from "../deps.ts"
import Collection = mongo.Collection
import Filter = mongo.Filter
export default function<type>(collection: Collection<type>, filter: Filter<type>) {
	return collection.findOne(filter, { noCursorTimeout: false })
}
```
- `const Collection = mongo.Collection` does not work, because const only copy `mongo.Collection`'s function constructor
- `type Collection = mongo.Collection` does not work if you want not only type but also the function constructor

**Comment by lts372005** [comment]
> > * `type Collection = mongo.Collection` also does not work, because `Generic type 'Collection<T>' requires 1 type argument(s)`
> >   => import is the only solution
> 
> Replace
> 
> ```ts
> import Collection = mongo.Collection
> import Filter = mongo.Filter
> ```
> 
> with
> 
> ```ts
> type Collection<T> = mongo.Collection<T>;
> type Filter<T> = mongo.Filter<T>;
> ```

Well this is awkward... it works...
I guess that will be the workaround

**Comment by lts372005** [comment]
Looks like that transpilation bug also cause denoland/deno_lint#934

**Comment by kitsonk** [comment]
@nayeemrmn thanks...  I guess we have to support it then.  Silly non-standard TypeScript syntax. 😢

**Comment by DmitryRomaniuk** [comment]
https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require
@lts372005 I think that syntax was discovered for cover require syntax in nodejs and as you write above it convert from import to var https://github.com/microsoft/TypeScript/issues/17
@kitsonk And I can cover that case in https://github.com/denoland/deno_lint/issues/934

**Comment by lucacasonato** [comment]
This might have been fixed in the latest swc pull in 1.15.3.

**Comment by kdy1** [comment]
Seems like already fixed on upstream.
